src/components/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -43,11 +43,7 @@ export const Breadcrumbs = ({
 
   const ref = React.useRef<HTMLUListElement>(null);
 
-  /**
-   * Needs useLayoutEffect over useEffect since it needs to be run before paint.
-   * Note: with React 18, might be able to use useEffect https://github.com/reactjs/reactjs.org/blob/d14cbdca2445cd676526c4c52e1e106342ff7bb3/content/docs/hooks-reference.md?plain=1#L155
-   */
-  React.useLayoutEffect(() => {
+  React.useEffect(() => {
```

Comment:
Does this solve the flakiness? The change seems fine regardless, but I'm a little skeptical.

The linked docs say effects run before paint only on user input or `flushSync`. Otherwise I think it's still after the browser paints.

Reply:
There's a case for changing the implementation here from `debounce` to `throttle`, and adjusting the timeout. In the storybook UI, switching viewports causes the window to ease in to the new size, and `debounce` waits the whole time to do any layout updates. I imagine the flakiness is still possible since the overall behavior on resize is still the same.

Reply:
Y'all are right, this won't solve the flakiness. Flakiness is happening in the FF snap as a result of the resize handler, which is fixed by adding a leading call to the debounce